### PR TITLE
Do not set port if port is None, required for HA mode

### PR DESCRIPTION
**Pull request by jettify**
`hdfs3` should have ability not to set port for HA mode, so `libhdfs3` can do right thing and use multiple namenodes. Here simple fix for this

Related issue: https://github.com/dask/hdfs3/issues/62

### CODE REVIEW

File: hdfs3/core.py [hunk 1]
```diff
@@ -174,18 +182,22 @@ def connect(self):
             return
 
         o = _lib.hdfsNewBuilder()
-        _lib.hdfsBuilderSetNameNodePort(o, self.port)
+        if port is not None:
```
Comment (mrocklin) [inline hdfs3/core.py]: Perhaps `if self.port is not None`

**Comment by mrocklin** [comment]
@martindurant any thoughts?

File: hdfs3/tests/test_hdfs3.py [hunk 2]
```diff
@@ -27,6 +28,7 @@ def hdfs():
 
     if hdfs.exists('/tmp/test'):
         hdfs.rm('/tmp/test')
+    hdfs.disconnect()
```
Comment (martindurant) [inline hdfs3/tests/test_hdfs3.py]: Why is this required, since `__del__` on the HDFileSystem will call disconnect anyway?

Comment (mrocklin) [inline hdfs3/tests/test_hdfs3.py]: Del is convenient, but typically untrusted.

**Comment by martindurant** [comment]
It all looks very reasonable to me, except that we could use an addition of about a paragraph in the docs about how to use HA. As far as I can tell, this code does a good job of not changing anything for those that aren't doing that.

File: hdfs3/tests/test_hdfs3.py (hunk 2 above)
Comment (martindurant) [inline hdfs3/tests/test_hdfs3.py]: I was wondering is @jettify had seen any specific problems here

Comment (jettify) [inline hdfs3/tests/test_hdfs3.py]: `__del__` is last resort, python encourage of use context managers instead half working destructor. As far as I remember `__del__` working as expected only since 3.4.1

Comment (jettify) [inline hdfs3/tests/test_hdfs3.py]: Here is more info http://stackoverflow.com/questions/18971451/how-many-times-can-del-be-called-per-object-in-python see Tim Peters answer.

**Comment by jettify** [comment]
@martindurant could you point me to the best place where I can add docs about HA mode?

**Comment by martindurant** [comment]
Most natural seems to be https://github.com/dask/hdfs3/blob/master/docs/source/hdfs.rst
I would add a new section "HDFS High-Availability".

**Comment by FlorianWilhelm** [comment]
I think this was an important fix for having HA with hdfs3. Would anyone care to make a new release?
